README.md

```diff
@@ -123,6 +142,141 @@ pre-commit install
+when it detects that the service has become unresponsive or inactive.
+It operates independently from the main service but works in tandem to ensure continuous live data reduction.
+
+
+### Watchdog Configuration
+
+The watchdog service reads its configuration from the same `/etc/livereduce.conf` file as the main service,
+but uses only a subset of settings specific to monitoring behavior.
+The watchdog-specific configuration is optional and uses sensible defaults if not specified.
+
+The watchdog configuration section supports the following optional keys:
+```json
+{
+  "watchdog": {
+    "interval": 60,
+    "threshold": 300
+  }
+}
+```
+
+Configuration parameters:
+- `watchdog.interval` (default: 60 seconds) - How often the watchdog checks the livereduce log file for activity.
+- `watchdog.threshold` (default: 300 seconds) - Maximum allowed time without log activity
+  before the watchdog considers the service unresponsive and triggers a restart. Must be at least 20 seconds.
+
+If the configuration file does not contain a `watchdog` section,
+the watchdog will use the default values shown above.
+Invalid values will trigger a warning and fall back to defaults.
+
+### Managing the watchdog service
+
+The watchdog service is managed independently of the main `livereduce` service using standard systemd commands:
+
+```shell
+sudo systemctl start livereduce_watchdog
+sudo systemctl stop livereduce_watchdog
+sudo systemctl restart livereduce_watchdog
+```
+
+Check the watchdog service status:
+```shell
+sudo systemctl status livereduce_watchdog
+```
+
+**Important operational considerations:**
+
+- The watchdog service starts **after** the `livereduce` service
+  (as defined by `After=livereduce.service` in the systemd unit).
+- **Stopping the watchdog does not stop the main `livereduce` service** - it only stops monitoring.
+  The main service will continue running without supervision.
+- **Restarting the watchdog does not restart `livereduce`**
+  unless the watchdog detects that the main service has become unresponsive.
+- The watchdog and main service must be managed separately.
+  Starting/stopping one does not automatically affect the other.
+- The watchdog service has `Restart=always` configured,
+  so systemd will automatically restart the watchdog if it crashes.
+
+### Watchdog Logging
+
+The watchdog maintains its own separate log file at `/var/log/SNS_applications/livereduce_watchdog.log`
+when run as the user `snsdata`.
+
+This log captures:
+- Watchdog startup and configuration validation messages
+- Detection of inactivity (when the main service log hasn't been updated within the threshold)
+- Restart actions taken against the main `livereduce` service
+- The last 20 lines of the main livereduce log at the time of restart (for correlation)
+- Status output from systemctl after triggering a restart
+
+To view the watchdog logs in real-time:
+```shell
+sudo tail -f /var/log/SNS_applications/livereduce_watchdog.log
```

Comment:
I thought that the log files were public (i.e. `644`). Does this really need sudo to view it?